src/NuGet.Core/NuGet.Common/ConcurrencyUtilities.cs

```diff
@@ -187,7 +187,7 @@ private static string BasePath
                     return _basePath;
                 }
 
-                _basePath = Path.Combine(NuGetEnvironment.GetFolderPath(NuGetFolderPath.Temp), "lock");
+                _basePath = Path.Combine(NuGetEnvironment.GetFolderPath(NuGetFolderPath.NuGetHome), "lock");
```

Comment:
This needs more design. Unanswered questions in my mind:

1. How will user of this latest nuget.exe clear this new lock file location?
1. How will we move to system-wide locking eventually?
1. Suppose we `$TMPDIR/NuGetScratch` to `$TMPDIR/NuGetScratch-$USERNAME` to get around the permissions issue. How will these be cleaned up in the future?